nemo/core/classes/modelPT.py

```diff
@@ -1301,6 +1301,7 @@ def cfg(self):
             Changes to this config are not reflected in the state of the model.
             Please create a new model using an updated config to properly update the model.
         """
+        self._set_hparams(OmegaConf.create({'cfg': self._cfg}))
```

Comment:
In order to always save the config. Otherwise the explicit `model.cfg = model.cfg` is required in order to preserve the latest config (well it may still be required, but best to preserve the config value during both get and set).

Reply:
It seems very strange to do this kind of update during a get.. And if we do, then why do we also still need that line? Having to do model.cfg = model.cfg is very unintuitive. Can we replace that with an explicit update_hparams method?

Reply:
Having yet another method to update configs seems redundant - we have the parameter in the class managing this. Sure it seems unintuitive but that's because there's no feasible way to 1) Give users a pointer to the config 2) let them modify in place 3) Update hyper parameters for wandb. 2 and 3 contradict each other.

We could take other approaches, as in explicitly calling this method inside of our "pre" train begin or whatever, but that would be outside the scope of this pr.